src/shared/util.js

```diff
@@ -827,46 +827,36 @@ class Util {
   }
 
   // Returns a rectangle [x1, y1, x2, y2] corresponding to the
-  // intersection of rect1 and rect2. If no intersection, returns 'false'
+  // intersection of rect1 and rect2. If no intersection, returns 'null'
   // The rectangle coordinates of rect1, rect2 should be [x1, y1, x2, y2]
   static intersect(rect1, rect2) {
-    function compare(a, b) {
-      return a - b;
-    }
+    // [a, b] inter [c d] != 0 <=> c <= b && d >= a
+    // [a, b] inter [c d] = [max(a, c), min(b, d)] with max <= min.
```

Comment:
It might just be a question about the formatting here, but I'm sorry to say that I'm actually finding this comment slightly less clear than just carefully reading the code directly :-)